pymongo/asynchronous/auth.py

```diff
@@ -161,7 +161,7 @@ def _password_digest(username: str, password: str) -> str:
     if len(password) == 0:
         raise ValueError("password can't be empty")
     if not isinstance(username, str):
-        raise TypeError("username must be an instance of str")
+        raise TypeError(f"username must be an instance of str, not {type(username)}.")
```

Comment:
It is customary to omit the period in these error messages. Do you mind changing these from `not {type(<>)}."` to `not {type(<>)}"`?

Reply:
Actually is seems there are examples of both in our codebase so either way is fine with me though I still have a slight preference for omitting it. @sleepyStick I'll defer to your opinion.

Reply:
I think I like omitting it as well, lemme clean it up and then I'll merge it :)